Core/include/Acts/Propagator/Navigator.hpp

```diff
@@ -46,10 +39,10 @@ struct NavigationOptions {
   /// always look for passive
   bool resolvePassive = false;
 
-  /// object to check against: at start
-  const object_t* startObject = nullptr;
-  /// object to check against: at end
-  const object_t* endObject = nullptr;
+  /// Hint for start object
+  const void* startObject = nullptr;
+  /// Hint for end object
+  const void* endObject = nullptr;
```

Comment:
Generally, we've tried to avoid bare void pointers in the code as much as possible. That doesn't mean to not ever use them, but to apply caution.

In the recent vertexing rewrite, I used a struct around a void pointer. The struct can be made to allow comparisons with arbitrary pointers, but the void pointer stays private. One can then also add a type index to ensure type safety, in case the pointer ever needs to be cast back to a concrete type. 

Would that be an option here?

Reply:
That would be an option or I could revive the template param. I guess it does not hurt that much